snykTask/src/index.ts

```diff
@@ -146,21 +153,39 @@ async function runSnykTest(
 
   const snykTestToolRunner = tl
     .tool(snykPath)
+    .argIf(taskArgs.testType == TestType.CODE, 'code')
+    .argIf(
+      taskArgs.dockerImageName || taskArgs.testType == TestType.CONTAINER_IMAGE,
+      'container',
+    )
     .arg('test')
     .argIf(
-      taskArgs.severityThreshold,
+      taskArgs.testType != TestType.CODE && taskArgs.severityThreshold,
       `--severity-threshold=${taskArgs.severityThreshold}`,
     )
-    .argIf(taskArgs.dockerImageName, `--docker`)
+    .argIf(
+      taskArgs.testType == TestType.CODE && taskArgs.codeSeverityThreshold,
+      `--severity-threshold=${taskArgs.codeSeverityThreshold}`,
+    )
     .argIf(taskArgs.dockerImageName, `${taskArgs.dockerImageName}`)
     .argIf(fileArg, `--file=${fileArg}`)
     .argIf(taskArgs.ignoreUnknownCA, `--insecure`)
     .argIf(taskArgs.organization, `--org=${taskArgs.organization}`)
     .argIf(taskArgs.projectName, `--project-name=${projectNameArg}`)
     .arg(`--json-file-output=${jsonReportOutputPath}`)
-    .argIf(isDebugMode(), '-d')
+    .argIf(isDebugMode() && taskArgs.testType != TestType.CODE, '-d')
```

Comment:
question: Why did you disable `-d` for code? I think it would be better to enable it for all products.

Reply:
this ties in with the approach used to send output from `snyk code test` to the `pipeExecOutputToTool` function. This will include debug analytics output which will break the `snyk-to-html` execution (invalid json).

Reply:
suggestion: It should not - debug output is created on standard error, command output on standard out. Please make it possible to get debug output for Snyk Code, as this is paramount for analysing issues.

Reply:
ok. an execution of `snyk code test --json` for its piped output file is only launched for prior successful execution with a non-existent json file and not for every `snyk code test` execution. in this way, debug with `-d` option is possible.